code/modules/mob/living/simple_animal/friendly/diona.dm

```diff
@@ -42,103 +43,99 @@
 		name = "[name] ([rand(1, 1000)])"
 		real_name = name
 
 	add_language("Rootspeak")
-	verbs += /mob/living/simple_animal/diona/proc/merge
+
+/mob/living/simple_animal/diona/UnarmedAttack(var/atom/A)
+	if(isdiona(A) && src in A.contents) //can't attack your gestalt
+		visible_message("[src] wiggles around a bit.")
+	else
+		..()
+
+/mob/living/simple_animal/diona/resist()
+	..()
+	split()
 
 /mob/living/simple_animal/diona/attack_hand(mob/living/carbon/human/M)
 	//Let people pick the little buggers up.
 	if(M.a_intent == INTENT_HELP)
 		if(isdiona(M))
 			to_chat(M, "You feel your being twine with that of [src] as it merges with your biomass.")
 			to_chat(src, "You feel your being twine with that of [M] as you merge with its biomass.")
-			verbs += /mob/living/simple_animal/diona/proc/split
-			verbs -= /mob/living/simple_animal/diona/proc/merge
+			throw_alert(gestalt_alert, /obj/screen/alert/nymph) //adds a screen alert that can call resist
 			forceMove(M)
 		else
 			get_scooped(M)
 
-	..()
-
-/mob/living/simple_animal/diona/proc/merge()
-	set category = "Diona"
-	set name = "Merge with gestalt"
-	set desc = "Merge with another diona."
-
-	if(iscarbon(loc))
-		verbs -= /mob/living/simple_animal/diona/proc/merge
-		return
+/mob/living/simple_animal/diona/proc/merge() //called by the hud (/obj/screen/diona/merge)
+	if(!isnymph(src) || src.stat)
+		return FALSE
 
 	var/list/choices = list()
-	for(var/mob/living/carbon/C in view(1,src))
-
-		if(!(Adjacent(C)) || !(C.client))
+	for(var/mob/living/carbon/human/H in view(1,src))
+		if(!(Adjacent(H)) || !isdiona(H))
 			continue
+		choices += H
 
-		if(ishuman(C))
-			var/mob/living/carbon/human/D = C
-			if(isdiona(D))
-				choices += C
+	if(!choices.len)
+		to_chat(src, "<span class='warning'>No suitable diona nearby.</span>")
 
 	var/mob/living/M = input(src,"Who do you wish to merge with?") in null|choices
 
-	if(!M || !src || !(Adjacent(M)))
+	if(!M || !src || !(Adjacent(M)) || src.stat) //input can take a while, so re-validate
 		return
 
-	if(ishuman(M))
+	if(isdiona(M))
 		to_chat(M, "You feel your being twine with that of [src] as it merges with your biomass.")
 		M.status_flags |= PASSEMOTES
-
 		to_chat(src, "You feel your being twine with that of [M] as you merge with its biomass.")
 		forceMove(M)
-		verbs += /mob/living/simple_animal/diona/proc/split
-		verbs -= /mob/living/simple_animal/diona/proc/merge
+		throw_alert("merged with gestalt", /obj/screen/alert/nymph) //adds a screen alert that can call resist
+		return TRUE
 	else
-		return
-
-/mob/living/simple_animal/diona/proc/split()
-	set category = "Diona"
-	set name = "Split from gestalt"
-	set desc = "Split away from your gestalt as a lone nymph."
-
-	if(!(iscarbon(loc)))
-		verbs -= /mob/living/simple_animal/diona/proc/split
-		return
-
+		return FALSE
+
+/mob/living/simple_animal/diona/proc/split() //called by the hud (obj/screen/alert/nymph)
+	if(!isnymph(src) || src.stat || !isdiona(loc))
```

Comment:
the `isnymph(src)` check does nothing here and should be removed. Also, `stat != CONSCIOUS`, not `src.stat`